Signal-Desktop: update to 5.0.0.

lane-brain opened the pull request:
<!-- Mark items with [x] where applicable -->

#### General
- [ ] This is a new package and it conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements)

#### Have the results of the proposed changes been tested?
- [x] I use the packages affected by the proposed changes on a regular basis and confirm this PR works for me
- [ ] I generally don't use the affected packages but briefly tested this PR

<!--
If GitHub CI cannot be used to validate the build result (for example, if the
build is likely to take several hours), make sure to
[skip CI](https://github.com/void-linux/void-packages/blob/master/CONTRIBUTING.md#continuous-integration).
When skipping CI, uncomment and fill out the following section.
Note: for builds that are likely to complete in less than 2 hours, it is not
acceptable to skip CI.
-->
<!-- 
#### Does it build and run successfully? 
(Please choose at least one native build and, if supported, at least one cross build. More are better.)
- [x] I built this PR locally for my native architecture, (ARCH-LIBC)
- [ ] I built this PR locally for these architectures (if supported. mark crossbuilds):
  - [ ] aarch64-musl
  - [ ] armv7l
  - [ ] armv6l-musl
-->

lane-brain commented:
Checked in with the maintainer @julio641742, they gave the go ahead to submit update PR

ericonr commented:
There's also #30505 

We need to figure out how to work with `git-lfs` here.

ericonr commented:
Are these patches imported from arch's packaging? https://github.com/archlinux/svntogit-community/tree/packages/signal-desktop/trunk

I think changing git config would be bad, but could you do `export HOME=$(mktemp -d)` in `pre_configure` and `rm -rf $HOME` in `post_install` ? I think it would work around the current hang up.

lane-brain commented:
The openssl linking and epoch source date patch yes. The former seemed justified because now we use openssl, the other one seemed necessitated by using tarballs instead of git. As far as I know about git-lfs, the documentation from Signal's github indicate that it's necessary for one of the yarn dependencies. 

As an aside, I fixed the linting error (vlicense).

And now the build has concluded but I ran into an error at post_install:

```
...
rm: cannot remove '/tmp/go/pkg/mod/golang.org/x/exp@v0.0.0-20200513190911-00229845015e/utf8string/string_test.go': Permission denied
rm: cannot remove '/tmp/go/pkg/mod/golang.org/x/exp@v0.0.0-20200513190911-00229845015e/winfsnotify/winfsnotify.go': Permission denied
rm: cannot remove '/tmp/go/pkg/mod/golang.org/x/exp@v0.0.0-20200513190911-00229845015e/winfsnotify/winfsnotify_test.go': Permission denied
=> ERROR: Signal-Desktop-5.0.0_1: post_install: 'rm -rf $HOME' exited with 1
=> ERROR:   in post_install() at srcpkgs/Signal-Desktop/template:65
```

ericonr commented on srcpkgs/Signal-Desktop/template:
```diff
@@ -1,26 +1,40 @@
+
+	# Have SQLCipher dynamically link from OpenSSL
+	# See https://github.com/signalapp/Signal-Desktop/issues/2634
+	patch --forward --strip=1 --input="${FILESDIR}/openssl-linking.patch"
+
+	# We can't read the release date from git so we use SOURCE_DATE_EPOCH instead
+	patch --forward --strip=1 --input="${FILESDIR}/expire-from-source-date-epoch.patch"
```
The patches should go in `patches` and left to be applied automatically. This is an arch artifact.

lane-brain replied:
believe me, I thought so too! but actually the files to be patched don't yet exist just after the extract, they need to be patched at this point

lane-brain replied:
I was able to leave out the epoch source date patch

ericonr commented:
`chmod -R +wX $HOME` works, but that's starting to sound like they build Go stuff in the middle of the build process??? Where do they even get the Go executable?

lane-brain commented:
> There's also #30505
> 
> We need to figure out how to work with `git-lfs` here.

by the way, here is the build error if we don't install git-lfs
```
gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
make: Entering directory '/builddir/Signal-Desktop-5.0.0/node_modules/@journeyapps/sqlcipher/build'
  CC(target) Release/obj.target/nothing/../../node-addon-api/nothing.o
  AR(target) Release/obj.target/../../node-addon-api/nothing.a
  COPY Release/nothing.a
  ACTION deps_sqlite3_gyp_action_before_build_target_unpack_sqlite_dep Release/obj/gen/sqlcipher-amalgamation-3033000/sqlite3.c
Traceback (most recent call last):
  File "/usr/lib/python3.9/tarfile.py", line 1682, in gzopen
    t = cls.taropen(name, mode, fileobj, **kwargs)
  File "/usr/lib/python3.9/tarfile.py", line 1659, in taropen
    return cls(name, mode, fileobj, **kwargs)
  File "/usr/lib/python3.9/tarfile.py", line 1522, in __init__
    self.firstmember = self.next()
  File "/usr/lib/python3.9/tarfile.py", line 2327, in next
    tarinfo = self.tarinfo.fromtarfile(self)
  File "/usr/lib/python3.9/tarfile.py", line 1112, in fromtarfile
    buf = tarfile.fileobj.read(BLOCKSIZE)
  File "/usr/lib/python3.9/gzip.py", line 300, in read
    return self._buffer.read(size)
  File "/usr/lib/python3.9/_compression.py", line 68, in readinto
    data = self.read(len(byte_view))
  File "/usr/lib/python3.9/gzip.py", line 487, in read
    if not self._read_gzip_header():
  File "/usr/lib/python3.9/gzip.py", line 435, in _read_gzip_header
    raise BadGzipFile('Not a gzipped file (%r)' % magic)
gzip.BadGzipFile: Not a gzipped file (b've')

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/builddir/Signal-Desktop-5.0.0/node_modules/@journeyapps/sqlcipher/deps/./extract.py", line 7, in <module>
    tfile = tarfile.open(tarball,'r:gz');
  File "/usr/lib/python3.9/tarfile.py", line 1629, in open
    return func(name, filemode, fileobj, **kwargs)
  File "/usr/lib/python3.9/tarfile.py", line 1686, in gzopen
    raise ReadError("not a gzip file")
tarfile.ReadError: not a gzip file
make: *** [deps/action_before_build.target.mk:13: Release/obj/gen/sqlcipher-amalgamation-3033000/sqlite3.c] Error 1
make: Leaving directory '/builddir/Signal-Desktop-5.0.0/node_modules/@journeyapps/sqlcipher/build'
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/builddir/Signal-Desktop-5.0.0/node_modules/node-gyp/lib/build.js:194:23)
gyp ERR! stack     at ChildProcess.emit (events.js:315:20)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:277:12)
gyp ERR! System Linux 5.11.18_1
gyp ERR! command "/usr/bin/node" "/builddir/Signal-Desktop-5.0.0/node_modules/node-gyp/bin/node-gyp.js" "build" "--build-from-source" "--module=/builddir/Signal-Desktop-5.0.0/node_modules/@journeyapps/sqlcipher/lib/binding/napi-v6-linux-x64/node_sqlite3.node" "--module_name=node_sqlite3" "--module_path=/builddir/Signal-Desktop-5.0.0/node_modules/@journeyapps/sqlcipher/lib/binding/napi-v6-linux-x64" "--napi_version=7" "--node_abi_napi=napi" "--napi_build_version=6" "--node_napi_label=napi-v6"
gyp ERR! cwd /builddir/Signal-Desktop-5.0.0/node_modules/@journeyapps/sqlcipher
gyp ERR! node -v v14.16.0
gyp ERR! node-gyp -v v6.1.0
gyp ERR! not ok 
node-pre-gyp ERR! build error 
node-pre-gyp ERR! stack Error: Failed to execute '/usr/bin/node /builddir/Signal-Desktop-5.0.0/node_modules/node-gyp/bin/node-gyp.js build --build-from-source --module=/builddir/Signal-Desktop-5.0.0/node_modules/@journeyapps/sqlcipher/lib/binding/napi-v6-linux-x64/node_sqlite3.node --module_name=node_sqlite3 --module_path=/builddir/Signal-Desktop-5.0.0/node_modules/@journeyapps/sqlcipher/lib/binding/napi-v6-linux-x64 --napi_version=7 --node_abi_napi=napi --napi_build_version=6 --node_napi_label=napi-v6' (1)
node-pre-gyp ERR! stack     at ChildProcess.<anonymous> (/builddir/Signal-Desktop-5.0.0/node_modules/node-pre-gyp/lib/util/compile.js:83:29)
node-pre-gyp ERR! stack     at ChildProcess.emit (events.js:315:20)
node-pre-gyp ERR! stack     at maybeClose (internal/child_process.js:1048:16)
node-pre-gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:288:5)
node-pre-gyp ERR! System Linux 5.11.18_1
node-pre-gyp ERR! command "/usr/bin/node" "/builddir/Signal-Desktop-5.0.0/node_modules/@journeyapps/sqlcipher/node_modules/.bin/node-pre-gyp" "install" "--build-from-source"
node-pre-gyp ERR! cwd /builddir/Signal-Desktop-5.0.0/node_modules/@journeyapps/sqlcipher
node-pre-gyp ERR! node -v v14.16.0
node-pre-gyp ERR! node-pre-gyp -v v0.15.0
node-pre-gyp ERR! not ok 
Failed to execute '/usr/bin/node /builddir/Signal-Desktop-5.0.0/node_modules/node-gyp/bin/node-gyp.js build --build-from-source --module=/builddir/Signal-Desktop-5.0.0/node_modules/@journeyapps/sqlcipher/lib/binding/napi-v6-linux-x64/node_sqlite3.node --module_name=node_sqlite3 --module_path=/builddir/Signal-Desktop-5.0.0/node_modules/@journeyapps/sqlcipher/lib/binding/napi-v6-linux-x64 --napi_version=7 --node_abi_napi=napi --napi_build_version=6 --node_napi_label=napi-v6' (1)
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
=> ERROR: Signal-Desktop-5.0.0_1: pre_build: 'yarn install --ignore-engines' exited with 1
=> ERROR:   in pre_build() at srcpkgs/Signal-Desktop/template:31
```

lane-brain commented:
> `chmod -R +wX $HOME` works, but that's starting to sound like they build Go stuff in the middle of the build process??? Where do they even get the Go executable?

No idea here. But unfortunately no this mktemp method runs into another issue:
```
=> Signal-Desktop-5.0.0_1: skipping check (XBPS_CHECK_PKGS is disabled) ...
=> Signal-Desktop-5.0.0_1: running pre-install hook: 00-libdir ...
=> Signal-Desktop-5.0.0_1: running pre-install hook: 02-script-wrapper ...
=> Signal-Desktop-5.0.0_1: running pre-install hook: 98-fixup-gir-path ...
=> Signal-Desktop-5.0.0_1: running do_install ...
=> Signal-Desktop-5.0.0_1: running post_install ...
=> Signal-Desktop-5.0.0_1: running post-install hook: 00-compress-info-files ...
=> Signal-Desktop-5.0.0_1: running post-install hook: 00-fixup-gir-path ...
=> Signal-Desktop-5.0.0_1: running post-install hook: 00-libdir ...
=> Signal-Desktop-5.0.0_1: running post-install hook: 00-uncompress-manpages ...
=> Signal-Desktop-5.0.0_1: running post-install hook: 01-remove-misc ...
=> Signal-Desktop-5.0.0_1: running post-install hook: 02-remove-libtool-archives ...
=> Signal-Desktop-5.0.0_1: running post-install hook: 02-remove-perl-files ...
=> Signal-Desktop-5.0.0_1: running post-install hook: 02-remove-python-bytecode-files ...
=> Signal-Desktop-5.0.0_1: running post-install hook: 03-remove-empty-dirs ...
=> Signal-Desktop-5.0.0_1: running post-install hook: 04-create-xbps-metadata-scripts ...
mktemp: failed to create file via template ‘/tmp/tmp.XXXXXXXXXX’: No such file or directory
/void-packages/common/hooks/post-install/04-create-xbps-metadata-scripts.sh: line 27: $tmpf: ambiguous redirect
/void-packages/common/hooks/post-install/04-create-xbps-metadata-scripts.sh: line 203: $tmpf: ambiguous redirect
/void-packages/common/hooks/post-install/04-create-xbps-metadata-scripts.sh: line 307: $tmpf: ambiguous redirect
/void-packages/common/hooks/post-install/04-create-xbps-metadata-scripts.sh: line 314: $tmpf: ambiguous redirect
/void-packages/common/hooks/post-install/04-create-xbps-metadata-scripts.sh: line 315: $tmpf: ambiguous redirect
   Added trigger 'gtk-icon-cache' for the 'INSTALL' script.
   Added trigger 'update-desktopdb' for the 'INSTALL' script.
/void-packages/common/hooks/post-install/04-create-xbps-metadata-scripts.sh: line 333: $tmpf: ambiguous redirect
/void-packages/common/hooks/post-install/04-create-xbps-metadata-scripts.sh: line 334: $tmpf: ambiguous redirect
/void-packages/common/hooks/post-install/04-create-xbps-metadata-scripts.sh: line 341: $tmpf: ambiguous redirect
/void-packages/common/hooks/post-install/04-create-xbps-metadata-scripts.sh: line 342: $tmpf: ambiguous redirect
/void-packages/common/hooks/post-install/04-create-xbps-metadata-scripts.sh: line 341: $tmpf: ambiguous redirect
/void-packages/common/hooks/post-install/04-create-xbps-metadata-scripts.sh: line 342: $tmpf: ambiguous redirect
/void-packages/common/hooks/post-install/04-create-xbps-metadata-scripts.sh: line 345: $tmpf: ambiguous redirect
/void-packages/common/hooks/post-install/04-create-xbps-metadata-scripts.sh: line 346: $tmpf: ambiguous redirect
/void-packages/common/hooks/post-install/04-create-xbps-metadata-scripts.sh: line 347: $tmpf: ambiguous redirect
/void-packages/common/hooks/post-install/04-create-xbps-metadata-scripts.sh: line 361: $tmpf: ambiguous redirect
/void-packages/common/hooks/post-install/04-create-xbps-metadata-scripts.sh: line 362: $tmpf: ambiguous redirect
mv: missing destination file operand after '/destdir//Signal-Desktop-5.0.0/INSTALL'
Try 'mv --help' for more information.
=> ERROR: Signal-Desktop-5.0.0_1: failed to write INSTALL metadata file!
```

notthewave commented:
this can be closed
